packages/@sanity/form-builder/src/sanity/DefaultAssetSource/utils/useMatchesMedia.ts

```diff
@@ -0,0 +1,25 @@
+import {useState, useCallback, useEffect} from 'react'
+import {useTheme} from '@sanity/ui'
+
+export function useMatchesMedia(
```

Comment:
Given that this is only used in one place, and resides within the `DefaultAssetSource` utils, is this a bit too generic/premature? There's something about the API here that I don't find too intuitive:

```ts
const isMobile = useMatchesMedia(1)
```
I had to go digging for the implementation to find out why `1` was used. I'd prefer to wait with this generalization and see if we can bake this into `@sanity/ui` or `@sanity/base`, instead inlining the check for now.

Reply:
I just found out that we have a util in Sanity UI called `useMediaIndex()` for this. It is kind of similar to the utility in this PR and gives us an index that maps to the breakpoints in Sanity UI.